### PR TITLE
Closes #3254: show checkbox while removing search engine 

### LINKED ISSUE #3254: Remove search engines doesn't work
## Steps to reproduce
1. Go to Settings>Search>Default search engine.
2. Tap the 3 dots button and select Remove.
3. Select the search engines you want to remove.

### Expected behavior
A checkbox is displayed for each engine to tap the ones you want to remove.

### Actual behavior
The checkboxes aren't displayed and can't select any of the installed engines.

### Device information
Google Pixel (Android 9)
LG Nexus 5 (Android 6.0.1)

* Focus version: 7.0

### PR DESCRIPTION
**Pull request by SubhrajyotiSen**
Also convert MultiselectSearchEngineListPreference to kotlin to override itemResId

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        defaultButton.isActivated = true
+    }
+
+    // Whenever an engine is checked or unchecked, we notify the menu
+    private fun bindEngineCheckboxesToMenu() {
+        for (i in 0 until searchEngineGroup!!.childCount) {
+            val engineButton = searchEngineGroup!!.getChildAt(i) as CompoundButton
+            engineButton.setOnCheckedChangeListener { _, _ ->
+                val context = context
+                if (context is ContextThemeWrapper) {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.kt]: I feel like this should be a global utility extension function on Context. We've repeated this pattern a lot in the code. Maybe Context.asActivity()?

Comment (SubhrajyotiSen) [reply app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.kt]: I found 2 such patterns. But how will the function actually work?
I am unable to understand how the utility function will handle the casting and after returning also execute the needed function call on the casted context

Comment (colintheshots) [reply app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.kt]: Here's the simplest version. Put this utility function in a util class and use it.

`fun Context.asActivity() = (this as? ContextThemeWrapper)?.baseContext as? Activity ?: this as? Activity`

`context?.asActivity()?.invalidateOptionsMenu()`

Comment (colintheshots) [reply app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.kt]: Using https://kotlinlang.org/docs/reference/typecasts.html#safe-nullable-cast-operator

**Comment by SubhrajyotiSen** [comment]
@colintheshots  I have updated the PR using the extension function

**Pull request merged**